ast.go

```diff
@@ -1977,3 +2065,85 @@ func (node *Revoke) walkSubtree(visit Visit) error {
 	}
 	return Walk(visit, node.Privileges, node.Table)
 }
+
+// resolvers
+
+func resolveReadStatementWalk(node Node, resolver ReadStatementResolver) (string, error) {
+	err := Walk(func(node Node) (bool, error) {
+		if funcExpr, ok := node.(*CustomFuncExpr); ok && funcExpr != nil {
+			resolvedString, err := resolveReadStatement(funcExpr, resolver)
+			if err != nil {
+				return true, err
+			}
+			funcExpr.ResolvedString = resolvedString
+		}
+		return false, nil
+	}, node)
+	if err != nil {
+		return "", err
```

Comment:
Nit: Maybe doing some wrapping of returned errors to provide more context on where errors happened?